vllm/v1/sample/rejection_sampler.py

```diff
@@ -144,26 +156,89 @@ def forward_native(
             accept_mask = (
                 output_token_ids[:, :-1] == draft_token_ids_tensor).cumprod(
                     dim=1)
+
+            # Identify valid positions (non-padding).
+            valid_mask = output_token_ids != INVALID_TOKEN_ID
+            # Generate mask with bonus token.
+            generate_mask = torch.cat([
+                accept_mask,
+                torch.zeros(accept_mask.size(0), 1, device=accept_mask.device)
+            ],
+                                      dim=1).to(torch.bool) & valid_mask
+            zeros_mask = (generate_mask == 0)
+            first_zero_idx = zeros_mask.float().argmax(dim=1)
+            # Figure out which rows actually contain at least one zero.
+            rows_with_zero = zeros_mask.any(dim=1)
+            # Use indexing to set the first zero in each of those rows to 1.
+            generate_mask[rows_with_zero, first_zero_idx[rows_with_zero]] = 1
+
+            output_token_ids[~generate_mask] = INVALID_TOKEN_ID
+
         else:
-            raise NotImplementedError(
-                "Currently, only greedy sampling is supported by "
-                "rejection sampler.")
-        # Identify valid positions (non-padding).
-        valid_mask = output_token_ids != INVALID_TOKEN_ID
-        # Generate mask with bonus token.
-        generate_mask = torch.cat([
-            accept_mask,
-            torch.zeros(accept_mask.size(0), 1, device=accept_mask.device)
-        ],
-                                  dim=1).to(torch.bool) & valid_mask
-        zeros_mask = (generate_mask == 0)
-        first_zero_idx = zeros_mask.float().argmax(dim=1)
-        # Figure out which rows actually contain at least one zero.
-        rows_with_zero = zeros_mask.any(dim=1)
-        # Use indexing to set the first zero in each of those rows to 1.
-        generate_mask[rows_with_zero, first_zero_idx[rows_with_zero]] = 1
-
-        output_token_ids[~generate_mask] = INVALID_TOKEN_ID
+            # Reference: https://arxiv.org/pdf/2211.17192
+            # 1. Extract the probabilities of the draft tokens.
+            # [batch_size, max_spec_len]
+            assert draft_probs is not None
```

Comment:
the assertion should cause the test_rejection_sampler to fail? https://github.com/vllm-project/vllm/pull/13933/files#diff-80ee7e2a62f9dcfbb8a312dc4e3948557e97ef187290daebbcae1e28596bda29R971